### PR TITLE
Replace Wiki API lookups with processed dumps

### PR DESCRIPTION
**Pull request by rmitsch**
## Goals

- Add functionality to download and process Wikipedia and Wikidata dumps.
- Replace Wikidata/Wikipedia API lookups with access to the processed dumps.

## Description

Adds some existing, slightly modified Wiki dump parsing routines from https://github.com/explosion/projects/tree/master/nel-wikipedia. Introduces a `WikiDataset` class stitching together those routines and functionality for efficient access/(de-)serialization of data extracted from raw dumps.

### Types of change

New feature.

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: benchmarks/nel/project.yml [hunk 1]
```diff
@@ -13,6 +13,14 @@ assets:
     url: 'https://zenodo.org/record/3970806/files/reddit_el.zip?download=1'
     checksum: 'eea345cc7574a5c9c376748d1871d557'
     description: Entity linking dataset scraped from Reddit. See [paper](https://arxiv.org/abs/2101.01228).
+  - dest: 'assets/wiki/wikidata_entity_dump.json.bz2'
+    url: 'https://dumps.wikimedia.org/wikidatawiki/entities/latest-all.json.bz2'
+    checksum: '42bc9dcb95adcef21121c04984412f97'
```
[the remark this answers is not in the document]
Comment (rmitsch) [reply benchmarks/nel/project.yml:18]: Yeah, I had a similar thought. We can also do this as a workflow step instead?

File: benchmarks/nel/project.yml [hunk 2]
```diff
@@ -52,6 +60,11 @@ commands:
     script:
       - "python -m spacy download ${vars.vectors_model}"
 
+  - name: parse_wiki_dumps
+    help: "Parses Wikipedia dumps. Note: This deletes any existing database file."
```
[the remark this answers is not in the document]
Comment (rmitsch) [reply benchmarks/nel/project.yml:64]: That's a good point, will adjust it (and yes, the parsing takes a long time).

File: benchmarks/nel/scripts/schemas.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    description: Optional[str] = Field(None, title="Full description.")
+    article_title: Optional[str] = Field(None, title="Article title.")
+    article_text: Optional[str] = Field(None, title="Article text.")
+
+
+class Annotation(BaseModel):
+    """Schema for single annotation."""
+
+    entity_name: str = Field(..., title="Entity name.")
+    entity_id: Optional[str] = Field(None, title="Entity name.")
```
[the remark this answers is not in the document]
Comment (rmitsch) [reply benchmarks/nel/scripts/schemas.py:26]: Yes.

File: benchmarks/nel/scripts/datasets/reddit.py [hunk 4]
```diff
@@ -96,11 +85,15 @@ def _create_annotated_docs(self) -> List[Doc]:
             file_names.append("posts.tsv")
         if self._options["comments"]:
             file_names.append("comments.tsv")
-        assert file_names, "Either 'posts' or 'comments' have to be True in corpus config."
+        assert (
+            file_names
+        ), "Either 'posts' or 'comments' have to be True in corpus config."
```
[the remark this answers is not in the document]
Comment (rmitsch) [reply benchmarks/nel/scripts/datasets/reddit.py:90]: It's a `black` thing... there is enough space to have everything in one line (and I did), but `black` doesn't like that :roll_eyes: Using a message variable now.

File: benchmarks/nel/scripts/schemas.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (svlandeg) [reply benchmarks/nel/scripts/schemas.py:26]: Then should we rename it? like why not have `entity_qid` everywhere? I know the old code didn't but it feels more clear, no?

File: benchmarks/nel/project.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (svlandeg) [reply benchmarks/nel/project.yml:18]: I think it's fine in "assets", but maybe just remove the checksum?

File: benchmarks/nel/scripts/schemas.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (svlandeg) [reply benchmarks/nel/scripts/schemas.py:26]: Or maybe it was named `entity_id` to be less specific to wikipedia/wikidata & more generic. I guess it's fine. But both `entity_name` and `entity_id` currently have the same "title" here in the `Field` declaration?

File: benchmarks/nel/scripts/wiki/wiki_dump_api.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+    return db_conn
+
+
+def parse(
+    db_conn: Optional[sqlite3.Connection] = None,
+    entity_config: Optional[Dict[str, Any]] = None,
+    article_text_config: Optional[Dict[str, Any]] = None,
+    alias_prior_prob_config: Optional[Dict[str, Any]] = None,
+) -> None:
+    """Parses Wikipedia and Wikidata dumps. Writes parsing results to file. Note that this takes hours.
```
Comment (svlandeg) [inline benchmarks/nel/scripts/wiki/wiki_dump_api.py:33]: Writes to a db, right?

**Pull request merged**